node/src/components/storage.rs

```diff
@@ -1,859 +1,578 @@
+/// On-disk storage configuration.
+#[derive(Clone, DataSize, Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// The path to the folder where any files created or read by the storage component will exist.
+    ///
+    /// If the folder doesn't exist, it and any required parents will be created.
+    pub(crate) path: PathBuf,
+    /// The maximum size of the database to use for the block store.
+    ///
+    /// The size should be a multiple of the OS page size.
+    max_block_store_size: usize,
+    /// The maximum size of the database to use for the deploy store.
+    ///
+    /// The size should be a multiple of the OS page size.
+    max_deploy_store_size: usize,
+    /// The maximum size of the database to use for the deploy store.
+    ///
+    /// The size should be a multiple of the OS page size.
+    max_deploy_metadata_store_size: usize,
 }
 
-#[allow(trivial_casts)]
-impl<B, D> StorageType for LmdbStorage<B, D>
-where
-    B: Value + WithBlockHeight + 'static,
-    D: Value + Item + 'static,
-{
-    type Block = B;
-    type Deploy = D;
-
-    fn new(config: WithDir<Config>) -> Result<Self> {
-        let root = config.with_dir(config.value().path());
-        fs::create_dir_all(&root).map_err(|error| Error::CreateDir {
-            dir: root.display().to_string(),
-            source: error,
-        })?;
-
-        let block_store_path = root.join(BLOCK_STORE_FILENAME);
-        let block_height_store_path = root.join(BLOCK_HEIGHT_STORE_FILENAME);
-        let deploy_store_path = root.join(DEPLOY_STORE_FILENAME);
-        let chainspec_store_path = root.join(CHAINSPEC_STORE_FILENAME);
-
-        let block_store = LmdbStore::new(block_store_path, config.value().max_block_store_size())?;
-        let block_height_store = LmdbBlockHeightStore::new(
-            block_height_store_path,
-            config.value().max_block_height_store_size(),
-        )?;
-        let deploy_store =
-            LmdbStore::new(deploy_store_path, config.value().max_deploy_store_size())?;
-        let chainspec_store = LmdbChainspecStore::new(
-            chainspec_store_path,
-            config.value().max_chainspec_store_size(),
-        )?;
-
-        Ok(LmdbStorage {
-            block_store: Arc::new(block_store),
-            block_height_store: Arc::new(block_height_store),
-            deploy_store: Arc::new(deploy_store),
-            chainspec_store: Arc::new(chainspec_store),
-        })
-    }
-
-    fn block_store(&self) -> Arc<dyn Store<Value = B>> {
-        Arc::clone(&self.block_store) as Arc<dyn Store<Value = B>>
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            // No one should be instantiating a config with storage set to default.
+            path: "/dev/null".into(),
+            max_block_store_size: 483_183_820_800,
+            max_deploy_store_size: 322_122_547_200,
+            max_deploy_metadata_store_size: 322_122_547_200,
```

Comment:
Frankly, I think we could without making these configurable altogether. They are ludicrously large (and should be), as there is almost no penalty for using large values for these. At the moment they are not even _used properly_.

They are also misleading, as they set the size of the memory map, only indirectly restricting disk usage. In general, we should scrap all these configuration options, they are a mirage.

Some background: https://github.com/lmdbjava/lmdbjava/issues/126#issuecomment-481058992

Reply:
As for the const thing, even if we kept this, I would want to resist publishing these constants elsewhere unless we need them and I am perfectly fine having them just in the `Default` impl.

Finally, comments like that suck, because they get outdated. Much better approach IMHO:

```
const GIGABYTE: usize = 1024 * 1024 * 1024

max_deploy_store_size: 450 * GIGABYTE
```

Make the compiler work for it!

Reply:
As to your first comment, that's not something we need to tackle in this PR.  Let's just maintain the current behaviour and discuss only the changes we need in order to get this PR completed.

> As for the const thing, even if we kept this, I would want to resist publishing these constants elsewhere unless we need them

I'm not asking for them to be public - they weren't public before and they don't need to be public now :)

> I am perfectly fine having them just in the Default impl.

That may be, but we have a long-standing policy of keeping magic numbers, strings and such like as consts, usually defined at the top of the mod/file.  So, let's just stick with our normal coding style please.

> Finally, comments like that suck, because they get outdated. Much better approach IMHO:

I'm quite happy with either comments or your preferred approach - as long as one of these options is chosen.

Reply:
> I'm not asking for them to be public - they weren't public before and they don't need to be public now :)

The only reason they were not public before is because of a technicality. If they were a type instead of a type alias they **would** have been public. We cannot have a public interface that uses private types.

I don't think duplicating the type alias in the tests is worthwhile at all here.